File: src/main/java/com/jhonycar/backend/entity/Veiculo.java
package com.jhonycar.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "veiculos")
public class Veiculo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String placa;

    @Column(nullable = false)
    private String modelo;

    @Column(nullable = false)
    private String marca;

    @Column(nullable = false)
    private Integer ano;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;
}

File: frontend/src/types/veiculo.ts
export interface Veiculo {
  id: number;
  placa: string;
  modelo: string;
  marca: string;
  ano: number;
  clienteId: number;
  clienteNome: string;
}

export interface VeiculoPayload {
  placa: string;
  modelo: string;
  marca: string;
  ano: number;
  clienteId: number;
}

export interface VeiculoModeloOption {
  id: number;
  marca: string;
  modelo: string;
  imagemUrl: string;
}

export interface VeiculoModeloPayload {
  marca: string;
  modelo: string;
  imagemUrl?: string;
  ativo?: boolean;
}

export interface VeiculoImagemUploadResponse {
  modeloId: number;
  imagemUrl: string;
}
